packages/@aws-cdk/toolkit/test/actions/watch.test.ts

```diff
@@ -144,6 +144,9 @@ describe('watch', () => {
     expect(deploySpy).toHaveBeenCalledWith(expect.anything(), 'watch', expect.objectContaining({
       cloudWatchLogMonitor: expect.anything(), // Not undefined
     }));
+
+    // Deactivate the cloudWatchLogMonitor that we created, otherwise the tests won't exit
+    (deploySpy.mock.calls[0][2] as any).cloudWatchLogMonitor?.deactivate();
```

Comment:
Should this be in a `finally` block then?

Reply:
We can only get a reference to the right object if the get far enough that the call succeeds.

So if we break this test the experience won't be great... but in that case we're broken anyway

Reply:
Ok, would be nice to make the broken experience better though. Won't insist.

Reply:
I don't think the way cloud watch logs monitoring is currently implemented makes a lot of sense for the toolkit. I'm currently making my way through the deploy action and can take a look.